include/oboe/Definitions.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OBOE_DEFINITIONS_H
+#define OBOE_DEFINITIONS_H
+
+#include <cstdint>
+#include <aaudio/AAudio.h>
+
+namespace oboe {
+
+    constexpr int32_t kUnspecified = 0;
+
+    constexpr int64_t kNanosPerMicrosecond =    1000;
+    constexpr int64_t kNanosPerMillisecond =    kNanosPerMicrosecond * 1000;
+    constexpr int64_t kMillisPerSecond =        1000;
+    constexpr int64_t kNanosPerSecond =         kNanosPerMillisecond * kMillisPerSecond;
+
+    enum class StreamState : aaudio_stream_state_t {
+        Uninitialized = AAUDIO_STREAM_STATE_UNINITIALIZED,
+        Open = AAUDIO_STREAM_STATE_OPEN,
+        Starting = AAUDIO_STREAM_STATE_STARTING,
+        Started = AAUDIO_STREAM_STATE_STARTED,
+        Pausing = AAUDIO_STREAM_STATE_PAUSING,
+        Paused = AAUDIO_STREAM_STATE_PAUSED,
+        Flushing = AAUDIO_STREAM_STATE_FLUSHING,
+        Flushed = AAUDIO_STREAM_STATE_FLUSHED,
+        Stopping = AAUDIO_STREAM_STATE_STOPPING,
+        Stopped = AAUDIO_STREAM_STATE_STOPPED,
+        Closing = AAUDIO_STREAM_STATE_CLOSING,
+        Closed = AAUDIO_STREAM_STATE_CLOSED,
+    };
+
+    enum class Direction : aaudio_direction_t {
+        Output = AAUDIO_DIRECTION_OUTPUT,
+        Input = AAUDIO_DIRECTION_INPUT,
+    };
+
+    enum class AudioFormat : aaudio_format_t {
+        Invalid = AAUDIO_FORMAT_INVALID,
+        Unspecified = AAUDIO_FORMAT_UNSPECIFIED,
+        I16 = AAUDIO_FORMAT_PCM_I16,
+        Float = AAUDIO_FORMAT_PCM_FLOAT,
+    };
+
+    enum class DataCallbackResult : aaudio_data_callback_result_t {
+        Continue = AAUDIO_CALLBACK_RESULT_CONTINUE,
+        Stop = AAUDIO_CALLBACK_RESULT_STOP,
+    };
+
+    enum class Result : aaudio_result_t {
```

Comment:
Consider renaming this class to "Error", which would allow dropping the "Error" prefix from the enum items.

Reply:
Is it normal to have an `Error::OK` value? We're going to see a lot of: 

`if (error != Error::OK){ `

Reply:
Yes, that's perhaps not the best expression. But seeing a lot of "Error" prefixes doesn't make me happy either. Although, I've just checked Vulkan's 'Result' type, and they also have "error" prefixes everywhere. OK, let's not consider changing anything here for now.